pixels/endpoints/moderation.py

```diff
@@ -0,0 +1,188 @@
+
+    records = await conn.fetch("SELECT * FROM users WHERE user_id=any($1::bigint[])", tuple(users))
+    db_users = [record["user_id"] for record in records]
+
+    non_db_users = set(users) - set(db_users)
+
+    # Ref:
+    # https://magicstack.github.io/asyncpg/current/faq.html#why-do-i-get-postgressyntaxerror-when-using-expression-in-1
+    await conn.execute("UPDATE users SET is_banned=TRUE WHERE user_id=any($1::bigint[])", db_users)
+
+    await conn.execute("UPDATE pixel_history SET deleted=TRUE WHERE user_id=any($1::bigint[])", db_users)
+
+    await request.state.canvas.sync_cache(conn, skip_check=True)
+
+    return ModBan(banned=db_users, not_found=list(non_db_users))
+
+
+@router.get("/pixel_history", response_model=t.Union[PixelHistory, Message])
+async def pixel_history(
+        request: Request,
+        x: int = Sizes.X_QUERY_VALIDATOR,
+        y: int = Sizes.Y_QUERY_VALIDATOR
+) -> t.Union[PixelHistory, Message]:
+    """Get the user who placed a specific pixel given its coordinates."""
+    sql = (
+        "SELECT user_id::text FROM pixel_history "
+        "WHERE x=$1 AND y=$2 AND NOT deleted "
+        "ORDER BY pixel_history_id DESC "
+        "LIMIT 1"
+    )
+    record = await request.state.db_conn.fetchrow(sql, x, y)
+
+    if not record:
+        return Message(message=f"No user history for pixel ({x}, {y})")
+
+    return PixelHistory(user_id=record["user_id"])
+
+
+@router.post("/webhook", response_model=Message)
+async def webhook(request: Request) -> Message:
+    """Send or update the Discord webhook image."""
+    last_message_id = await request.state.redis_pool.get("last-webhook-message")
+
+    now = datetime.now()
+
+    # Generate payload that will be sent in payload_json
+    data = {
+        "content": "",
+        "embeds": [{
+            "title": "Pixels State",
+            "image": {
+                "url": f"attachment://pixels_{now.timestamp()}.png"
+            },
+            "footer": {
+                "text": "Last updated"
+            },
+            "timestamp": now.isoformat()
+        }]
+    }
+
+    # Run Pillow stuff in executor because these actions are blocking
+    loop = asyncio.get_event_loop()
+    image = await loop.run_in_executor(
+        None,
+        partial(
+            Image.frombytes,
+            "RGB",
+            (Sizes.WIDTH, Sizes.HEIGHT),
+            bytes(await request.state.canvas.get_pixels())
+        )
+    )
+
+    # Increase size of image so this looks better in Discord
+    image = await loop.run_in_executor(
+        None,
+        partial(
+            image.resize,
+            Sizes.WEBHOOK_SIZE,
+            Image.NEAREST
+        )
+    )
+
+    # BytesIO gives a file-like interface for saving
+    # and later this is able to get actual content what will be sent.
+    file = io.BytesIO()
+    await loop.run_in_executor(None, partial(image.save, file, format="PNG"))
+
+    # Name file to pixels.png
```

Comment:
Not sure how to get this meaning better across
```suggestion
    # Name file to pixels_TIMESTAMP.png
```